omero2pandas/io_tools.py

```diff
@@ -0,0 +1,149 @@
+# encoding: utf-8
+#
+# Copyright (c) 2025 Glencoe Software, Inc. All rights reserved.
+#
+# This software is distributed under the terms described by the LICENCE file
+# you can find at the root of the distribution bundle.
+# If the file is missing please request a copy by contacting
+# support@glencoesoftware.com.
+import os
+import io
+import logging
+
+import omero
+from omero.rtypes import unwrap
+from omero.gateway import BlitzGateway
+
+LOGGER = logging.getLogger(__name__)
+
+
+class OriginalFileIO(io.RawIOBase):
+    """
+    Reader for loading data from OMERO OriginalFile objects with a
+    Python file-like interface
+
+    Only a single RawFileStore instance can exist per BlitzGateway,
+    loading multiple readers will change the target file for the FileStore.
+
+    For optimal performance this class should be wrapped in a BufferedReader,
+    use the get_annotation_reader convenience function for this.
+    """
+    def __init__(self, conn: BlitzGateway, file_id: int, reporter=None):
+        LOGGER.debug(f"Creating new  for file {file_id}")
+        self._prx = conn.createRawFileStore()
+        self._file_id = file_id
+        self.open()
+        if not (size := self._prx.size()):
+            raise omero.ClientError(f"Invalid size for OriginalFile {file_id}")
+        self._size = size
+        self._offset = 0
+        self._reporter = reporter
+        if self._reporter is not None:
+            reporter.reset(total=size)
+
+    def __enter__(self):
+        self.open()
```

Comment:
If `open()` is called in `__init__` I don't think it needs to be called here as well.